icefall/lexicon.py

```diff
@@ -0,0 +1,66 @@
+import logging
+import re
+from pathlib import Path
+from typing import List
+
+import k2
+import torch
+
+
+class Lexicon(object):
+    """Phone based lexicon.
+
+    TODO: Add BpeLexicon for BPE models.
+    """
+
+    def __init__(
+        self, lang_dir: Path, disambig_pattern: str = re.compile(r"^#\d+$")
```

Comment:
I think that `__init__` should be changed to be able to construct a Lexicon from existing symbol tables in program memory, which also makes it easier to test this class, e.g.:

```python
# Construct arbitrary object
lexicon = Lexicon(phones=phone_symbol_table, words=word_symbol_table, L=L_fst)

# Construct from a lang dir
lexicon = Lexicon.from_dir(‘data/lang’)

# Store in a lang dir
lexicon.to_dir(‘data/lang’)
```